### DISCUSSION TITLE
Custom interface drawer not showing when field is clicked

### DISCUSSION THREAD

**Discussion by jsx101** [directus/directus#16446] [Archive]
I'm trying to make a custom interface with a drawer to select items from a related collection (like the one below for Many-To-One interface) but cannot get the DrawerCollection component to appear.
![image](https://user-images.githubusercontent.com/40978303/201647762-e31f87f2-1a81-4630-a307-f91c5d2e2fde.png)

Here is my code. What am I doing wrong?

```
<template>
	<v-input
		clickable
		:placeholder="'Select an item...'"
		@click="onPreviewClick"
	>
		<v-icon class="expand" name="expand_more" />
	</v-input>

	<drawer-collection 
		:active="true"
		:collection="relatedCollectionRef"
		:selection="selection"
		@input="onSelection"
	/>
	
	<div>Collection items: {{items}}</div>
	<div>Selection: {{selection}}</div>
	<div>Collection: {{collection}}</div>
	<div>Field: {{field}}</div>
</template>

<script>
import { useItems, useStores } from '@directus/extensions-sdk';
import { computed } from '@vue/reactivity';
import { ref } from 'vue';


export default {
	props: {
		value: {
			type: String,
			default: null,
		},
		collection: {
			type: String
		},
		field: {
			type: String
		},
		fieldsToDisplay: {
			type: Array,
			default: ['id']
		},
		template: {
			type: String,
			default: null
		},
		enableCreate: {
			type: Boolean,
			default: true
		},
		enableSelect: {
			type: Boolean,
			default: true
		}
	},
	emits: ['input'],
	setup(props, { emit }) {
		const {collection, field, fieldsToDisplay} = props;
		const { useRelationsStore } = useStores();
		const relationsStore = useRelationsStore();
		const relationInfo = relationsStore.getRelationForField(collection, field);
		const relatedCollection = relationInfo.related_collection;
		const relatedCollectionRef = ref(relatedCollection);
		const fieldsToDisplayRef = ref(fieldsToDisplay)
		const items = useItems(relatedCollectionRef, {
			fields: fieldsToDisplayRef
		});

		const selectModalActive = ref(false);
		
		function onPreviewClick() {
			selectModalActive.value = true;
			return;
		}
		function onSelection() {
			selectModalActive.value = false;
		}

		const selection = computed(() => {
			return items.items;
		});
	

		return { items, relatedCollectionRef, selectModalActive, selection, onPreviewClick, onSelection };
		
	},
};
</script>
```

And here is the UI in the form. `Company` field is the M2O field and the other is my custom interface. Nothing happens when clicking my custom interface field but the borders do light up the same way `Company` does
![image](https://user-images.githubusercontent.com/40978303/201649380-b47a7952-e556-4155-bd63-1c74473b26f0.png)

The drawer collection element in my interface is showing up in Inspect Element like this 
![image](https://user-images.githubusercontent.com/40978303/201650543-8e761337-6fc9-446f-b660-17fa3e923e89.png)

**Comment by azrikahar** [comment]
This is due to `<drawer-collection>` not being one of the exposed components as mentioned in #7949.

There's another discussion about this in #13139 for more info/reasoning, but the OP there does mention it's possible copy the component over to your extension and use it locally in the meantime 👍

**Reply by jsx101** [reply]
Got it, thanks. 

I saw this https://github.com/directus/directus/discussions/15744 and assumed I can use DrawerCollection too without importing like the user used DrawerItem, or did I misunderstand the answer?
